File: pb/src/main/java/org/sqlbuilder/pb/foreign/FnAlias.java
package org.sqlbuilder.pb.foreign;

import org.sqlbuilder.pb.objects.Word;
import org.sqlbuilder.pb.objects.RoleSet;

import java.util.Comparator;
import java.util.HashSet;
import java.util.Set;

public class FnAlias extends Alias
{
	public static final Comparator<FnAlias> COMPARATOR = Comparator //
			.comparing(FnAlias::getPbRoleSet) //
			.thenComparing(FnAlias::getPbWord) //
			.thenComparing(FnAlias::getRef) //
			.thenComparing(FnAlias::getPos);

	public static final Set<FnAlias> SET = new HashSet<>();

	// C O N S T R U C T O R

	public static FnAlias make(final String clazz, final String pos, final RoleSet pbRoleSet, final Word word)
	{
		var a = new FnAlias(clazz, pos, pbRoleSet, word);
		SET.add(a);
		return a;
	}

	private FnAlias(final String clazz, final String pos, final RoleSet pbRoleSet, final Word word)
	{
		super(clazz, pos, pbRoleSet, word);
	}
}

File: pb31/src/main/java/org/sqlbuilder/pb/foreign/FnAlias.kt
package org.sqlbuilder.pb.foreign

import org.sqlbuilder.pb.objects.RoleSet
import org.sqlbuilder.pb.objects.Word

class FnAlias private constructor(clazz: String, pos: String, pbRoleSet: RoleSet, word: Word) : Alias(clazz, pos, pbRoleSet, word) {
    companion object {

        val COMPARATOR: Comparator<FnAlias> = Comparator
            .comparing<FnAlias, RoleSet> { it.pbRoleSet}
            .thenComparing {it.pbWord}
            .thenComparing {it.ref}
            .thenComparing {it.pos}

        val SET: MutableSet<FnAlias> = HashSet<FnAlias>()

        fun make(clazz: String, pos: String, pbRoleSet: RoleSet, word: Word): FnAlias {
            val a = FnAlias(clazz, pos, pbRoleSet, word)
            SET.add(a)
            return a
        }
    }
}
